### ISSUE TITLE
setState clears out the editor

### ISSUE THREAD

**Issue by Firebrand** [securingsincity/react-ace#551]
Hi guys,

Great component. I'm having a problem where whenever I run setState the contents of Ace Editor are completely wiped out.

Simple example:

```
  state = {
    buttonText: "Click here"
  };

  buttonPressed = () => {
    this.setState({ buttonText: "Clicked" });
  };

  render() {
    return (
      <React.Fragment>
        <button onClick={this.buttonPressed}>{this.state.buttonText}</button>
        <AceEditor mode="javascript" theme="monokai" name="editor" />
      </React.Fragment>
    );
  }
}
```

**Comment by PatrickNausha** [comment]
Having the same problem when using the `defaultValue` prop. It looks like react-ace is defaulting the `value` prop to `''`:
https://github.com/securingsincity/react-ace/blob/020e2bd7492d2b2de5f7ecac7b8972f10030a93b/src/ace.js#L361
It uses that prop on every update:
https://github.com/securingsincity/react-ace/blob/020e2bd7492d2b2de5f7ecac7b8972f10030a93b/src/ace.js#L153

It seems like if you don't use the `value` prop, the default behavior is to update the editor contents to `''` every render.

I suppose the workaround is to only use react-ace as a controlled component and always provide the `value` prop.

**Comment by securingsincity** [comment]
The state will not be maintained in the example you have there. Yes, you would need to use it as a controlled component and pass the value in when you update it.

**Comment by securingsincity** [comment]
This should be solved in 9.1.0
